java/client/src/org/openqa/selenium/firefox/GeckoDriverService.java

```diff
@@ -186,7 +192,8 @@ protected GeckoDriverService createDriverService(File exe, int port,
                                                      ImmutableList<String> args,
                                                      ImmutableMap<String, String> environment) {
       try {
-        GeckoDriverService service = new GeckoDriverService(exe, port, args, environment);
+        Duration timeout = (firefoxBinary == null) ? Duration.ofSeconds(45) : Duration.ofMillis(firefoxBinary.getTimeout());
```

Comment:
You change the default timeout from 20 to 45 seconds here. I realise that the `XpiDriverService` uses the timeout from the binary (which is 45), but it's preferable to not change behaviour here.